### ISSUE TITLE
Support less typish configuration situations

### ISSUE THREAD

**Issue by treeowl** [ekmett/hyperloglog#13]
Is there a good argument against providing something like this?

```haskell
coerceConfig :: forall p q . (HLL.ReifiesConfig p, HLL.ReifiesConfig q) =>
                HLL.HyperLogLog p -> Maybe (HLL.HyperLogLog q)
coerceConfig hll
    | HLLC.reflectConfig (Proxy :: Proxy p) == HLLC.reflectConfig (Proxy :: Proxy q) =
        Just $ unsafeCoerce hll
    | otherwise = Nothing
```

**Comment by treeowl** [comment]
Also in that code, I think `Config` deserves a custom `Eq` instance, recognizing that the first field is sufficient, or at least seems to be. It might also be good to offer a serializable global configuration version constant somewhere.

**Comment by treeowl** [comment]
Fixed by #15
